### PR TITLE
Address sphinx documentation warnings

### PR DESCRIPTION
**Pull request by jodygarnett**
This PR acts as a safety measure on further documentation changes, and is useful to catch causal sphinx errors (often introduced when editing directly from github).

* [x] introduce use of `-W --keep-going` to enforce warnings
* [x] address sphinx documentation warnings

I found two pages to be rather troubled:

* `csw-configuration.rst` defined several references (used as link targets from the inspire page). Sphinx only allows references before a heading (as the heading text is used the link is used in another document.
  
  ```
  .. _reference:
  
  heading
  -------- 
  ```
  
  I introduced some headings to address these warnings, and had to work on page order so the result would make sense.  

* The page `creating-custom-editor.rst` had several code examples that broken due to indentation. The page structure also made use of nested bullet points to group content, and had a few references without headings.

I would prefer if this PR was reviewed and merged based on changes to the document structure. While there may be inaccuracies (or clarifications) desired for the document content ... they are not the subject of this PR.

### CODE REVIEW

**Comment by josegar74** [comment]
I have tried the PR with `mvn clean package` and get the following:

```
Building documentation in html for 'en' 'fr'
set -e; for lang in 'en' 'fr'; do echo "Building language $lang"  && make html -e SPHINXOPTS="-D language='$lang'" -e "BUILDDIR='target/doc/$lang' -d target/doctrees"; done
Building language en
sphinx-build -b html -d 'target/doc/en' -d target/doctrees/doctrees  -D language='en' source -W --keep-going -j auto --no-color -q 'target/doc/en' -d target/doctrees/html
/private/tmp/doc/source/tutorials/hookcustomizations/ui/index.rst:57: WARNING: Could not lex literal_block as "javascript". Highlighting skipped.
/private/tmp/doc/theme/search.html:20: RemovedInSphinx30Warning: To modify script_files in the theme is deprecated. Please insert a <script> tag directly in your theme instead.
  {{ super() }}
make[1]: *** [html] Error 1
make: *** [html_languages] Error 2
[ERROR] Command execution failed.
```

Without the changes in the pull request, the build works.

**Comment by jodygarnett** [comment]
Nice, what version of the Sphinx are you running an you trying `make html`

Aside: I am concerned how to fix any French warnings/errors.

File: source/customizing-application/editor-ui/creating-custom-editor.rst [hunk 1]
```diff
@@ -1,911 +1 @@
-.. _creating-custom-editor:Customizing editor##################.. _creating-custom-editor-editor:
-        
-A metadata editor configuration is defined for a specific schema plugin standard
```
Comment (fxprunayre) [inline source/customizing-application/editor-ui/creating-custom-editor.rst:3]: Better to fix formatting issue in here https://github.com/geonetwork/core-geonetwork/blob/master/schemas/config-editor.xsd#L30 as this doc page is created from it. See https://geonetwork-opensource.org/manuals/trunk/en/contributing/writing-documentation.html#building-the-standards-docs

**Comment by juanluisrp** [comment]
I (and GN website Docker image) is getting this error:
```
Building documentation in html for 'en' 'fr'
set -e; for lang in 'en' 'fr'; do echo "Building language $lang"  && make html -e SPHINXOPTS="-D language='$lang'" -e "BUILDDIR='target/doc/$lang' -d target/doctrees"; done
Building language en
make[1]: Entering directory '/geonetwork-doc'
sphinx-build -b html -d 'target/doc/en' -d target/doctrees/doctrees  -D language='en' source -W --keep-going -j auto --no-color -q 'target/doc/en' -d target/doctrees/html
/geonetwork-doc/source/tutorials/hookcustomizations/ui/index.rst:57: WARNING: Could not lex literal_block as "javascript". Highlighting skipped.
make[1]: *** [Makefile:82: html] Error 1
make[1]: Leaving directory '/geonetwork-doc'
make: *** [Makefile:69: html_languages] Error 2
[ERROR] Command execution failed.
org.apache.commons.exec.ExecuteException: Process exited with an error: 2 (Exit value: 2)
```

```
sphinx-build --version
sphinx-build 3.0.1
```

```
root@0682c4135ead:/geonetwork-doc# pip3 list
Package                       Version
----------------------------- ----------
alabaster                     0.7.12
asn1crypto                    0.24.0
Babel                         2.8.0
certifi                       2020.4.5.1
chardet                       3.0.4
click                         7.1.2
cryptography                  2.6.1
distro-info                   0.21
docutils                      0.16
entrypoints                   0.3
idna                          2.9
imagesize                     1.2.0
Jinja2                        2.11.2
JSTools                       1.0
keyring                       17.1.1
keyrings.alt                  3.1.1
MarkupSafe                    1.1.1
packaging                     20.4
Pillow                        7.1.2
pip                           20.1.1
pycrypto                      2.6.1
pycurl                        7.43.0.2
Pygments                      2.6.1
PyGObject                     3.30.4
pyparsing                     2.4.7
python-apt                    1.8.4.1
python-Levenshtein            0.12.0
python-slugify                1.2.6
pytz                          2020.1
pyxdg                         0.25
requests                      2.23.0
SecretStorage                 2.3.1
setuptools                    40.8.0
six                           1.12.0
snowballstemmer               2.0.0
Sphinx                        3.0.1
sphinx-bootstrap-theme        0.7.1
sphinx-intl                   2.0.1
sphinx-rtd-theme              0.4.3
sphinxcontrib-applehelp       1.0.2
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        1.0.3
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.4
transifex-client              0.13.9
unattended-upgrades           0.1
Unidecode                     1.1.1
urllib3                       1.25.9
wheel                         0.32.3
```

**Comment by juanluisrp** [comment]
```
/geonetwork-doc/source/tutorials/hookcustomizations/ui/index.rst:57: WARNING: Could not lex literal_block as "javascript". Highlighting skipped.
```
Can be fixed changing `javascript` to `less` in `source/tutorials/hookcustomizations/ui/index.rst:57`

**Comment by jodygarnett** [comment]
Thanks for the feedback and testing everyone! Ends up it was my fault, I reset my python environment and was using `Sphinx-1.8.4`!

**Comment by jodygarnett** [comment]
Fixed code example, tested against sphinxdoc 3.1.1, squashed into a single commit.

**Comment by jodygarnett** [comment]
If this is working for all involved can it be merged please. Rude kind of PRs get out of date very quickly as docs are updated.

**Comment by jodygarnett** [comment]
@juanluisrp can you confirm your concerns have been addressed please.

**Pull request merged**